Game/RainMeadow.GameHooks.cs

```diff
@@ -180,6 +182,16 @@ private void Room_LoadedCheck(On.Room.orig_Loaded orig, Room self)
             }
         }
 
+        private void RoomSettings_ctor(On.RoomSettings.orig_ctor orig, RoomSettings self, string name, Region region, bool template, bool firstTemplate, SlugcatStats.Name playerChar)
+        {
+            if (isStoryMode(out var storyGameMode))
+            {
+                playerChar = storyGameMode.currentCampaign;
```

Comment:
In case Turtle wants also wants to take a look at this, I'll spare him some decompiling time:

```
        if (template)
        {
            filePath = AssetManager.ResolveFilePath("World" + Path.DirectorySeparatorChar + region.name + Path.DirectorySeparatorChar + name + ".txt");
        }
        else if (!File.Exists(filePath = ((playerChar == null) ? "" : WorldLoader.FindRoomFile(name, includeRootDirectory: false, "_settings-" + playerChar.value + ".txt"))))
        {
            string path = ((!ModManager.MSC || !name.EndsWith("-2")) ? WorldLoader.FindRoomFile(name, includeRootDirectory: false, "_settings.txt") : WorldLoader.FindRoomFile(name.Substring(0, name.Length - 2), includeRootDirectory: false, "-2_settings.txt"));
            if (File.Exists(path))
            {
                filePath = path;
            }
            else if (name.EndsWith("-2"))
            {
                filePath = WorldLoader.FindRoomFile(name.Substring(0, name.Length - 2), includeRootDirectory: false, "_settings.txt");
            }
            else
            {
                filePath = path;
            }
        }
```

Also of interest, this class uses Save() and Load() methods that take in a filepath that's assigned during the ctor. 

@youbitchoc the screenshot looks good. If you can verify that on death / sleep the world is loaded correctly that would be great.

Reply:
as for placement of code, GameHooks would be more appropriate imo.

Reply:
tested to work across cycles and deaths as well as having clients load the room